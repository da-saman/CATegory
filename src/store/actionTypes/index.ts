export const GET_ALL_CATEGORIES_REQUESTED = 'GET_ALL_CATEGORIES_REQUESTED';
export const GET_ALL_CATEGORIES_SUCCEEDED = 'GET_ALL_CATEGORIES_SUCCEEDED';
export const GET_ALL_CATEGORIES_FAILED = 'GET_ALL_CATEGORIES_FAILED';
export const GET_CAT_IMAGES_REQUESTED = 'GET_CAT_IMAGES_REQUESTED';
export const GET_CAT_IMAGES_SUCCEEDED = 'GET_CAT_IMAGES_SUCCEEDED';
export const GET_CAT_IMAGES_FAILED = 'GET_CAT_IMAGES_FAILED';