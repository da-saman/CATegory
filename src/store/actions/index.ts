import {
  GET_ALL_CATEGORIES_SUCCEEDED,
  GET_ALL_CATEGORIES_REQUESTED,
  GET_CAT_IMAGES_SUCCEEDED,
  GET_CAT_IMAGES_REQUESTED
} from '../actionTypes';


export const fetchCategories = () => ({
  type: GET_ALL_CATEGORIES_REQUESTED
})
export const storeCategories = (categories: any) => ({
  type: GET_ALL_CATEGORIES_SUCCEEDED,
  payload: {
    categories: categories
  }
});

export const fetchCatImages = (id: number) => ({
  type: GET_CAT_IMAGES_REQUESTED,
  payload: {
    id: id
  }
})
export const storeTheCategory = (category: any) => ({
  type: GET_CAT_IMAGES_SUCCEEDED,
  payload: {
    category: category
  }
})