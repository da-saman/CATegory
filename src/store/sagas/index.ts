import { all } from "redux-saga/effects";

import fetchAllCategoriesSaga from "./fetchCategories";
import fetchCatImagesSaga from "./fetchCatImages";

export default function* rootSaga(): any {
  yield all([fetchAllCategoriesSaga(), fetchCatImagesSaga()]);
}
