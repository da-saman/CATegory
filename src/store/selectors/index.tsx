export const allCategoriesSelector = (store: any) => store.categories;

export const catImagesSelector = (store: any) => store.catImages;

export const pageSelector = (store: any) => store.page;