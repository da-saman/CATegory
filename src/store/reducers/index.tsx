import { GET_ALL_CATEGORIES_SUCCEEDED, GET_THE_CATEGORY_SUCCEEDED } from '../actionTypes';
const initialState = {
  categories: [{ "id": 5, "name": "boxes" }, { "id": 15, "name": "clothes" }],
  cats: [],
  selectedCategory: null
}


const reducer = function (state: any = initialState, action: any) {
  switch (action.type) {
    case GET_ALL_CATEGORIES_SUCCEEDED: {
      const { categories } = action.payload;
      return {
        ...state,
        categories: categories
      }
    }
    case GET_THE_CATEGORY_SUCCEEDED: {
      const { theCategory} = action.payload;
      return { 
        ...state,
        selectedCategory: theCategory
      }
    }
    default: {
      return state;
    }
  }
}

export default reducer;