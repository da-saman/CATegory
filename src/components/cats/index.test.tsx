import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { initialState } from "../../store/reducers";
import { MemoryRouter } from "react-router";
import Cats from "./index";
import { fetchCatImages } from "../../store/actions";

const mockStore = configureStore([]);

describe("My Connected React-Redux Component", () => {
  let store: any;
  let component: any;

  beforeEach(() => {
    store = mockStore(initialState);
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/cats/2"]}>
          <Cats />
        </MemoryRouter>
      </Provider>
    );
  });

  it("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should dispatch an action on mount", () => {
    renderer.act(() => {
      component.root.findByType("button");
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    // expect(store.dispatch).toHaveBeenCalledWith([fetchCatImages(2, false),fetchCatImages(2, true)]);
  });
});
