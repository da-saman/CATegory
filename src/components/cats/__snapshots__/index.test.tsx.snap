// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="sc-bdnxRM gTseO"
>
  <div
    className="sc-gtsrHT eXrpdh"
  />
  <button
    className="sc-hKFxyN iSomKC"
    onClick={[Function]}
  >
    Load More
  </button>
</div>
`;
